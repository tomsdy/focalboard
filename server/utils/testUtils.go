package utils

import "github.com/stretchr/testify/mock"

var Anything = mock.MatchedBy(func(interface{}) bool { return true })
