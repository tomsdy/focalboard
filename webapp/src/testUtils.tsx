// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import {IntlProvider} from 'react-intl'
import React from 'react'

export const wrapIntl = (children: any) => <IntlProvider locale='en'>{children}</IntlProvider>

export function mockDOM(): void {
    window.focus = jest.fn()
    document.createRange = () => {
        const range = new Range()

        range.getBoundingClientRect = jest.fn()

        range.getClientRects = () => {
            return {
                item: () => null,
                length: 0,
                [Symbol.iterator]: jest.fn(),
            }
        }

        return range
    }
}
