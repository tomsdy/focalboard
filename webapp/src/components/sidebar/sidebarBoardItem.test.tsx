// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import React from 'react'

import {createMemoryHistory} from 'history'
import {Router} from 'react-router-dom'

import {render} from '@testing-library/react'

import {Provider as ReduxProvider} from 'react-redux'

import configureStore from 'redux-mock-store'

import {TestBlockFactory} from '../../test/testBlockFactory'

import {wrapIntl} from '../../testUtils'

import SidebarBoardItem from './sidebarBoardItem'

describe('components/sidebarBoardItem', () => {
    const board = TestBlockFactory.createBoard()

    const view = TestBlockFactory.createBoardView(board)
    view.fields.sortOptions = []
    const history = createMemoryHistory()

    const categoryBlocks1 = TestBlockFactory.createCategoryBlocks()
    categoryBlocks1.name = 'Category 1'
    categoryBlocks1.blockIDs = [board.id]

    const categoryBlocks2 = TestBlockFactory.createCategoryBlocks()
    categoryBlocks2.name = 'Category 2'

    const categoryBlocks3 = TestBlockFactory.createCategoryBlocks()
    categoryBlocks3.name = 'Category 3'

    const allCategoryBlocks = [
        categoryBlocks1,
        categoryBlocks2,
        categoryBlocks3,
    ]

    const state = {
        users: {
            me: {
                id: 'user_id_1',
            },
        },
        boards: {
            current: board.id,
            boards: {
                [board.id]: board,
            },
        },
        views: {
            current: view.id,
            views: {
                [view.id]: view,
            },
        },
        teams: {
            current: {
                id: 'team-id',
            },
        },
    }

    test('sidebar board item', () => {
        const mockStore = configureStore([])
        const store = mockStore(state)

        const component = wrapIntl(
            <ReduxProvider store={store}>
                <Router history={history}>
                    <SidebarBoardItem
                        categoryBlocks={categoryBlocks1}
                        board={board}
                        allCategories={allCategoryBlocks}
                        isActive={true}
                        showBoard={jest.fn()}
                        showView={jest.fn()}
                        onDeleteRequest={jest.fn()}
                    />
                </Router>
            </ReduxProvider>,
        )
        const {container} = render(component)
        expect(container).toMatchSnapshot()
    })
})
