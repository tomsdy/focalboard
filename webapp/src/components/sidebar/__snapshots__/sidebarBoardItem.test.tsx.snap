// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/sidebarBoardItem sidebar call hideSidebar 1`] = `
<div>
  <div
    class="SidebarBoardItem"
  >
    <div
      class="octo-sidebar-item ' expanded "
    >
      <button
        class="Button IconButton"
        type="button"
      >
        <svg
          class="DisclosureTriangleIcon Icon"
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polygon
            points="37,35 37,65 63,50"
          />
        </svg>
      </button>
      <div
        class="octo-sidebar-title"
        title="board title"
      >
        i board title
      </div>
      <div
        aria-label="menuwrapper"
        class="MenuWrapper"
        role="button"
      >
        <button
          class="Button IconButton"
          type="button"
        >
          <i
            class="CompassIcon icon-dots-horizontal OptionsIcon"
          />
        </button>
      </div>
    </div>
    <div
      class="octo-sidebar-item subitem "
    >
      <svg
        class="BoardIcon Icon"
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          height="80"
          rx="5"
          ry="5"
          width="80"
          x="10"
          y="10"
        />
        <polyline
          points="28,25 28,55"
          style="stroke-width: 15px;"
        />
        <polyline
          points="50,25 50,70"
          style="stroke-width: 15px;"
        />
        <polyline
          points="72,25 72,45"
          style="stroke-width: 15px;"
        />
      </svg>
      <div
        class="octo-sidebar-title"
        title="view title"
      >
        view title
      </div>
    </div>
    <div
      class="octo-sidebar-item subitem "
    >
      <svg
        class="BoardIcon Icon"
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          height="80"
          rx="5"
          ry="5"
          width="80"
          x="10"
          y="10"
        />
        <polyline
          points="28,25 28,55"
          style="stroke-width: 15px;"
        />
        <polyline
          points="50,25 50,70"
          style="stroke-width: 15px;"
        />
        <polyline
          points="72,25 72,45"
          style="stroke-width: 15px;"
        />
      </svg>
      <div
        class="octo-sidebar-title"
        title="view title"
      >
        view title
      </div>
    </div>
  </div>
</div>
`;
